crates/router/src/core/payments/transformers.rs

```diff
@@ -707,7 +707,7 @@ where
                         .set_incremental_authorization_allowed(
                             payment_intent.incremental_authorization_allowed,
                         )
-                        .set_fingerprint(payment_intent.fingerprint_id)
+                        .set_fingerprint(payment_attempt.fingerprint_id)
```

Comment:
```suggestion
                        .set_fingerprint(payment_intent.fingerprint_id)
```
The fingerprint in the response comes from the payment intent. We send back the fingerprint only if the payment is successful (which is the case if the fingerprint is put into payment intent)